app/src/main/java/org/fossasia/openevent/app/event/detail/EventDetailFragment.java

```diff
@@ -81,7 +81,8 @@ public void onCreate(Bundle savedInstanceState) {
         if (arguments != null)
             initialEventId = arguments.getLong(EVENT_ID);
 
-        eventDetailPresenter.attach(initialEventId, this);
+        eventDetailPresenter.attach(this);
+        eventDetailPresenter.attachKey(initialEventId);
```

Comment:
This has to repeat in each class, which makes it automatically an anti-pattern. Clearly, this always needs to be called together and thus remain in a single method